xds/src/test/java/io/grpc/xds/XdsSdsClientServerTest.java

```diff
@@ -87,6 +92,34 @@ public void plaintextClientServer() throws IOException, URISyntaxException {
     assertThat(unaryRpc("buddy", blockingStub)).isEqualTo("Hello buddy");
   }
 
+  @Test
+  public void fallbackProtocolNegotiator_expectException() throws IOException, URISyntaxException {
```

Comment:
i am not sure what this is testing. it provides fallbackPN that throws "no fallback negotiation" exception?

Reply:
The test makes sure the fallback PN is invoked in an easy way. I can see the contradiction of the fallback PN throwing exception saying "no fallback exception". I can change the text message. In connection with the other thread: if fallback PN is now optional, I can use the same fallback PN (as internal default) that throws an exception? Or else the HandlerPickerHandler can generate an error when there is no fallback PN. Which is better?

Reply:
mock PN can be better fit here. `verify(mockFallbackPn).newHandler();`.
The thrown exception should be able to converted to meaningful `Status`, probably `UNAVAILABLE`. the conversion logic is in `WriteBufferingAndExceptionHandler#exceptionCaught`. you can do it in default FallbackPN or `HandlerPickerHandler` whichever makes more sense.

Reply:
Just used `null` and verified that server is shutdown because of missing fallback PN